src/google-contetnt-provider.ts

```diff
@@ -0,0 +1,232 @@
+    .bucket(bucketName)
+    .file(fileName)
+    .getMetadata();
+
+  console.log(`File: ${metadata.name}`);
+  console.log(`Bucket: ${metadata.bucket}`);
+  console.log(`Storage class: ${metadata.storageClass}`);
+  console.log(`Self link: ${metadata.selfLink}`);
+  console.log(`ID: ${metadata.id}`);
+  console.log(`Size: ${metadata.size}`);
+  console.log(`Updated: ${metadata.updated}`);
+  console.log(`Generation: ${metadata.generation}`);
+  console.log(`Metageneration: ${metadata.metageneration}`);
+  console.log(`Etag: ${metadata.etag}`);
+  console.log(`Owner: ${metadata.owner}`);
+  console.log(`Component count: ${metadata.component_count}`);
+  console.log(`Crc32c: ${metadata.crc32c}`);
+  console.log(`md5Hash: ${metadata.md5Hash}`);
+  console.log(`Cache-control: ${metadata.cacheControl}`);
+  console.log(`Content-type: ${metadata.contentType}`);
+  console.log(`Content-disposition: ${metadata.contentDisposition}`);
+  console.log(`Content-encoding: ${metadata.contentEncoding}`);
+  console.log(`Content-language: ${metadata.contentLanguage}`);
+  console.log(`Media link: ${metadata.mediaLink}`);
+  console.log(`KMS Key Name: ${metadata.kmsKeyName}`);
+  console.log(`Temporary Hold: ${metadata.temporaryHold}`);
+  console.log(`Event-based hold: ${metadata.eventBasedHold}`);
+  console.log(
+    `Effective Expiration Time: ${metadata.effectiveExpirationTime}`
+  );
+  console.log(`Metadata: ${metadata.metadata}`);
+}
+get(fileName).catch(console.error);
+
+/**
+* Updates a file.
+ * @param storage
+ * @param bucketName
+ */
+export async function update() {
+  throw new Error("Not implemented");
+}
+
+/**
+Uploads the file to the bucket
+* @param storage
+* @param bucketName
+* @param filePathLocal
+* @returns {Promise<void>}
+*/
+export async function save(storage : any, bucketName: string, filePathLocal: string) {
+  // Uploads a local file to the bucket
+  try{
+  storage.bucket(bucketName).upload(filePathLocal, {
+    // Support for HTTP requests made with `Accept-Encoding: gzip`
+    gzip: true,
+    metadata: {
+      cacheControl: 'no-cache',
+    },
+  });
+  console.log(`${filePathLocal} uploaded to ${bucketName}.`);
+}
+catch{
+  
+}
+}
+save(storage,bucketName,filePathLocal).catch(console.error);
+
+/**
+Deletes the file from the bucket
+* @param storage
+* @param bucketName
+* @param fileName
+* @returns {Promise<void>}
+*/
+export async function remove(fileName : string) {
+  try{
+  storage.bucket(bucketName).file(fileName).delete();
+  console.log(`gs://${bucketName}/${fileName} deleted.`);
+  }
+  catch{
+    
+  }
+}
+remove(fileName).catch(console.error);
+
+/**
+ Delete current bucket
+ * @param storage
+ * @param bucketName
+ * @returns {Promise<void>}
+*/
+export async function removeBucket( bucketName : string) {
+  // Deletes the bucket
+  try{
+  storage.bucket(bucketName).delete();
+  console.log(`Bucket ${bucketName} deleted.`);
+  }
+  catch{
+
+  }
+}
+export class GoogleContentProvider{
```

Comment:
This class should have the methods that implement the IContentProvider interface.